### PR TITLE
Pass raw request args to handshake

### PR DESCRIPTION
**Pull request by nateReiners**
### _Summary_
**What**
- This PR makes a change to pass raw request args to our signer selection and handshake methods
**Why**
- To enable smart wallet onboarding optimization through the passing of request params

### _How did you test your changes?_
Locally
Unit Tests

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:43]: why await?

Comment (nateReiners) [reply packages/wallet-sdk/src/CoinbaseWalletProvider.ts]: wow you're fast. good catch!

removed [b490a3d](https://github.com/coinbase/coinbase-wallet-sdk/pull/1392/commits/b490a3ddd01002c46a0a0719f1bfdc7fe0341379)

File: packages/wallet-sdk/src/core/message/RPCMessage.ts
Comment (bangtoven) [inline packages/wallet-sdk/src/core/message/RPCMessage.ts:17]: what's this?

File: packages/wallet-sdk/src/sign/scw/SCWSigner.ts
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/scw/SCWSigner.ts:65]: https://github.com/coinbase/coinbase-wallet-sdk/pull/1388#discussion_r1750031176
thanks @talentlessguy !
image

File: packages/wallet-sdk/src/core/message/RPCMessage.ts
Comment (nateReiners) [reply packages/wallet-sdk/src/core/message/RPCMessage.ts]: whoops was added during a bad stash apply - removed [8d488f4](https://github.com/coinbase/coinbase-wallet-sdk/pull/1392/commits/8d488f4103a32ba6f86cbe16b2aaaf60dfaa0228)[8d488f4](https://github.com/coinbase/coinbase-wallet-sdk/pull/1392/commits/8d488f4103a32ba6f86cbe16b2aaaf60dfaa0228)[8d488f4](https://github.com/coinbase/coinbase-wallet-sdk/pull/1392/commits/8d488f4103a32ba6f86cbe16b2aaaf60dfaa0228)

File: packages/wallet-sdk/src/core/message/RPCMessage.ts
Comment (bangtoven) [inline packages/wallet-sdk/src/core/message/RPCMessage.ts:18]: dude. you gotta self review your PR before asking another approval @nateReiners

Comment (nateReiners) [reply packages/wallet-sdk/src/core/message/RPCMessage.ts]: [6ab13a0](https://github.com/coinbase/coinbase-wallet-sdk/pull/1392/commits/6ab13a09fb821d8df468c41d6019886c17e9825c) 

My bad, tried to move to fast! Looks good to me now 🙏

**Review by bangtoven — APPROVED** [review]

**Pull request merged**
